app/views/user/profile.php

```diff
@@ -24,9 +24,21 @@
         <div class="profile-info">
             <!-- Profile Picture -->
             <div class="profile-picture">
-
+                <?php if ($userId !== $profileUserId): ?>
+                    <!-- Only show follow/unfollow buttons if viewing another user's profile -->
+                    <?php if (isset($isFollowing) && $isFollowing): ?>
+                        <form action="/user/unfollow/<?= htmlspecialchars($profileUserId); ?>" method="POST">
+                            <button type="submit" class="btn btn-danger unfollow">Unfollow</button>
+                        </form>
+                    <?php else: ?>
+                        <form action="/user/follow/<?= htmlspecialchars($profileUserId); ?>" method="POST">
+                            <button type="submit" class="btn btn-primary follow">Follow</button>
+                        </form>
+                    <?php endif; ?>
+                <?php endif; ?>
```

Comment:
Why did you move it to the top again ?

Reply:
Makes more sense as the button feels so out of place elsewhere.

You'll see when you see the changes